doc/SAI-IPv6-Segment-Routing-Update.md

```diff
@@ -0,0 +1,484 @@
+ *        and replaced by sai_local_sid_entry_endpoint_flavor_t**
+ */
+
+typedef enum _sai_next_hop_endpoint_pop_type_t
+{
+    /** Penultimate segment pop */
+    SAI_NEXT_HOP_ENDPOINT_POP_TYPE_PSP,
+
+    /** Ultimate Segment pop */
+    SAI_NEXT_HOP_ENDPOINT_POP_TYPE_USP,
+
+} sai_next_hop_endpoint_pop_type_t;
+
+
+
+/**
+ * @brief Attribute id for next hop
+ */
+typedef enum _sai_next_hop_attr_t
+{
+...
+...
+
+    /**
+     * @brief Next hop entry Segment Route SID List
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_SEGMENTROUTE_SIDLIST
+     * @condition SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_SEGMENTROUTE_SIDLIST
+     */
+    SAI_NEXT_HOP_ATTR_SEGMENTROUTE_SIDLIST_ID,
+
+    /**
+     * @brief Next hop entry Segment Route Endpoint Function. **Deprecated and replaced
+     * with SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE.**
+     *
+     * @type sai_next_hop_endpoint_type_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @condition SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_SEGMENTROUTE_ENDPOINT
+    ** * @deprecated true**
+     */
+    SAI_NEXT_HOP_ATTR_SEGMENTROUTE_ENDPOINT_TYPE,
+
+    /**
+     * @brief Next hop entry Segment Route Endpoint Pop Option. **Deprecated and replaced
+     * with SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_FLAVOR**
+     *
+     * @type sai_next_hop_endpoint_pop_type_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @condition SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_SEGMENTROUTE_ENDPOINT
+     *** @deprecated true**
+     */
+    SAI_NEXT_HOP_ATTR_SEGMENTROUTE_ENDPOINT_POP_TYPE,
+
+...
+...
+
+} sai_next_hop_attr_t;
+```
+
+### sairouterinterface.h ###
+
+1. Added attribute to enable/disable SRv6 on a RIF.
+
+```   
+typedef enum _sai_router_interface_attr_t
+{
+   ...
+   
+    /**
+     * @brief Admin SRV6 state
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_ROUTER_INTERFACE_ATTR_ADMIN_SRV6_STATE,
+
+    ...
+
+} sai_router_interface_attr_t;
+```
+
+
+### saisegmentroute.h ###
+
+1. Added new enums for SR Headend functionality with reduced SRH. These new behaviors reduce the length of 
+   SRH by excluding the first segment in the pushed IPv6 header. First segment is placed in only the DA of 
+   the outer IPv6 header.
+
+```
+   typedef enum _sai_segmentroute_sidlist_type_t
+{
+    ...
+
+    /** Insertion of Segment Route Policy with Reduced SRH */
+    SAI_SEGMENTROUTE_SIDLIST_TYPE_INSERT_RED,
+
+    ...
+
+    /** Encapsulation in a Segment Route Policy with Reduced SRH */
+    SAI_SEGMENTROUTE_SIDLIST_TYPE_ENCAPS_RED,
+
+
+} sai_segmentroute_sidlist_type_t;
+``` 
+
+2. Added enum sai_local_sid_entry_endpoint_type_t to provide the initial list of endpoint behaviors.
+
+```
+/**
+ * @brief Enum defining Endpoint Action types
+ */
+typedef enum _sai_local_sid_entry_endpoint_type_t
```

Comment:
Moreover: I am afraid that use of "local_sid" term going forward may lead to confusion - recall that in SRv6 micro-segment (uSID) draft-filsfils-spring-net-pgm-extension-srv6-usid, there are segments are global (uN) and other as local (uA, uD). A endpoint node terminates both its global and local segments.

Shall we just call it "sid_entry" instead of "local_sid_entry" here and other places?

Reply:
Fair point. I used the term "local_sid" because it's being used in other implementations like VPP and Linux. Also, net-pgm document used to have a MyLocalSID table but I don't see it there anymore. Do you think "my_sid_entry" will be a more appropriate name since this table is supposed to have all the SIDs advertised by this node?